### PR TITLE
Repeated slashes should be ignored in file scheme

### PR DESCRIPTION
**Pull request by watilde**
Fixes https://github.com/whatwg/url/issues/232.

***
[Preview](https://s3.amazonaws.com/pr-preview/watilde/url/feature/check-file-slashes.html) | [Diff](https://s3.amazonaws.com/pr-preview/whatwg/url/6688baa..watilde:feature/check-file-slashes:b46bf40.html)

### CODE REVIEW

**Comment by annevk** [comment]
Are you planning on writing tests as well? There's some indentation issues with that patch, but it looks reasonable overall. Did you verify it with Node.js's implementation?

(I think you mean to write "ignored" rather than "arranged".)

**Comment by watilde** [comment]
@annevk Yeah I've verified on [the branch](https://github.com/watilde/node/commit/247f960f7125c1dfd24599a3800117d19e4ccb55) I posted in the issue report, and it worked. I just created [PR](https://github.com/w3c/web-platform-tests/pull/4762) for WPT now, then I will fix the indentation issues and update the commit message with the PR link.

File: url.bs [hunk 1]
```diff
@@ -1561,11 +1563,30 @@ string <var>input</var>, optionally with a <a>base URL</a> <var>base</var>, opti
...
+
+            <ol>
+             <li><p>If <a>remaining</a> does not start with "<code>//</code>",
+             <a>syntax violation</a>.
+
+             <li><p>Set <var>state</var> to <a>file state</a>.
+            </ol>
+          <li>
+            <p>Otherwise, set <var>state</var>
+            to <a>authority state</a>, and decrease <var>pointer</var> by one.</p>
```
Comment (domenic) [inline url.bs]: Running the jsdom/whatwg-url tests, this pointer decrease needs to happen even if the scheme is file.

Comment (watilde) [inline url.bs]: Ahh that's right! I will update it.

File: url.bs [hunk 2]
```diff
@@ -1561,11 +1562,33 @@ string <var>input</var>, optionally with a <a>base URL</a> <var>base</var>, opti
...
+        <p>If <var>url</var>'s <a for=url>scheme</a> is "<code>file</code>", run these
+        subsubsteps:
+        <ol>
+         <li>
+          <p>If <a>remaining</a> does not start with "<code>//</code>",
+          <a>syntax violation</a>.</p>
+         </li>
+         <li>
+          <p>Set <var>state</var> to <a>file state</a>,
+          and decrease <var>pointer</var> by one.</p>
```
Comment (domenic) [inline url.bs]: You can factor out the pointer decrease. Let me push a commit to show how.

**Pull request closed without merging**